pkg/cmd/dotgit/compress.go

```diff
@@ -0,0 +1,59 @@
+package dotgit
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/MaineK00n/vuls-data-update/pkg/dotgit/compress"
+)
+
+func newCmdCompress() *cobra.Command {
+	options := &struct {
+		output string
+		root   string
+
+		level int
+	}{
+		output: "",
+		root:   "",
+
+		level: 3,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "compress <repository path>",
+		Short: "Compress vuls-data-* dotgit with Zstandard",
+		Example: heredoc.Doc(`
+			$ vuls-data-update dotgit compress vuls-data-raw-debian-security-tracker
+			$ vuls-data-update dotgit compress vuls-data-raw-debian-security-tracker-archive-1 --output /tmp/vuls-data-raw-debian-security-tracker.tar.zst --root vuls-data-raw-debian-security-tracker 
+		`),
+		Args: cobra.ExactArgs(1),
+		PreRunE: func(_ *cobra.Command, args []string) error {
+			if options.output == "" {
+				options.output = fmt.Sprintf("%s.tar.zst", filepath.Base(args[0]))
+			}
+
+			if options.root == "" {
+				options.root = filepath.Base(args[0])
+			}
+
+			return nil
+		},
+		RunE: func(_ *cobra.Command, args []string) error {
+			if err := compress.Compress(args[0], options.output, compress.WithRoot(options.root), compress.WithLevel(options.level)); err != nil {
+				return errors.Wrap(err, "failed to dotgit compress")
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.output, "output", "o", options.output, "output to a specified file")
+	cmd.Flags().StringVarP(&options.root, "root", "", options.root, "root directory name when compressing")
+	cmd.Flags().IntVarP(&options.level, "level", "#", options.level, "compression level (1-22)")
```

Comment:
Just curiousity, want to know # is chosen.

Reply:
```console
zstd --help
*** zstd command line interface 64-bits v1.4.8, by Yann Collet ***
Usage : 
      zstd [args] [FILE(s)] [-o file] 

FILE    : a filename 
          with no FILE, or when FILE is - , read standard input
Arguments : 
 -#     : # compression level (1-19, default: 3)
...
```

Reply:
Zstd's # is the instruction to replace `#` with some numbers. 

```
% zstd -3 go.mod
go.mod               : 37.59%   (  3.37 KiB =>   1.27 KiB, go.mod.zst)
% zstd -# 3 go.mod
Incorrect parameters
```

Reply:
https://github.com/MaineK00n/vuls-data-update/compare/7af1a3036cb3967da15afd13b82519a1feb0dda1..36a77984578bdd27bc6bc97bcf75f0bcaee4ae04